### PR TITLE
Added task to maintain translations

### PR DESCRIPTION
**Pull request by mister-ben**
## Description
An grunt task to hopefully help keep translations up to date. Compares each language file against the keys in en.json.

Missing translations are outputted to the console and a table in a new doc at docs/translations-needed.md is updated.

`grunt check-translations`

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: build/tasks/languages.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      let missing = [];
+      for (const string in source) {
+        if (!target[string]) {
+          grunt.log.writeln(`${filename} missing "${string}"`);
+          missing.push(string);
+        }
+      }
+      if (missing.length > 0) {
+        grunt.log.error(`${filename} is missing ${missing.length} translations.`);
+        table += `|${filename}|${missing.join('|\n| |')}|\n`;
```
Comment (gkatsev) [inline build/tasks/languages.js]: maybe add the missing length after the `filename`?
As it is now it makes it seem a bit like `(5)` and `(13)` need translations.

**Comment by gkatsev** [comment]
This is pretty awesome.

File: build/tasks/languages.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module.exports = function(grunt) {
+  grunt.registerTask('check-translations', 'Check that translations are up to date', function(){
+    const source = require('../../lang/en.json');
+    let doc = grunt.file.read('docs/translations-needed.md');
+    let table = '|---|---|\n';
```
Comment (gkatsev) [inline build/tasks/languages.js]: might be better to build out the table using something like https://github.com/wooorm/markdown-table

File: build/tasks/languages.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      if (missing.length > 0) {
+        grunt.log.error(`${filename} is missing ${missing.length} translations.`);
+        table += `|${filename}|${missing.join('|\n| |')}|\n`;
+        table += `| | (${missing.length}) |\n`;
+      } else {
+        grunt.log.ok(`${filename} is up to date.`);
+        table += `|${filename}|(Complete)|\n`;
+      }
+    });
+    doc = doc.replace(/\|---\|---\|\n(.*\n)+/, table);
```
Comment (gkatsev) [inline build/tasks/languages.js]: might be safer to use something like https://github.com/thlorenz/update-section and do something like what doctoc does for specifying the location: https://github.com/thlorenz/doctoc#specifying-location-of-toc

**Comment by gkatsev** [comment]
Though, I guess having something is better than nothing and we can always make the task better later on.

**Comment by mister-ben** [comment]
Improved the table generation and replacement.

**Comment by gkatsev** [comment]
Awesome, thanks @mister-ben!
LGTM

**Comment by misteroneill** [comment]
Very cool! LGTM for sure!

**Comment by gkatsev** [comment]
@mister-ben can you rebase from master again? It'll have the `Audio Track` addition, then we won't need to re-run this task again after merging.

**Comment by gkatsev** [comment]
Awesome, thanks!

**Comment by gkatsev** [comment]
Haha, oops, just realized that this will need to be updated as part of https://github.com/videojs/video.js/pull/3576. Totally should've merge that earlier.
I'll make a PR for that update after merging.

**Pull request closed without merging**